site/docs/guides/create-derivation.md

```diff
@@ -0,0 +1,251 @@
+
+   Your draft is set up, but still exists only on the Estuary servers.
+
+5. Pull the draft locally to edit the specification files.
+
+   ```console
+   flowctl draft develop
+   ```
+
+   The specification files are written to your working directory.
+
+## Add a derivation
+
+1. Locate the specification YAML file for the collection you want to transform.
+
+   In your working directory, you'll see a top-level file called `flow.yaml`.
+   Within a subdirectory that shares the name of your Data Flow, you'll find a second `flow.yaml` — this contains the collection specification.
+
+   :::tip
+   Use `tree` to visualize your current working directory. This is a helpful tool to understand the [files that underlie your local draft](../concepts/flowctl.md#development-directories).
+   :::
+
+2. Open the specification file in your preferred editor.
+
+   It will look similar to the following (this example uses the default collection from the Hello World test capture, available in the web app):
+
+   ```yaml
+   collections:
+      estuary/data-flow-name/greetings:
+         schema:
+            properties:
+            count:
+               type: integer
+            message:
+               type: string
+            required:
+            - count
+            - message
+            type: object
+         key:
+            - /count
+    ```
+
+   You'll add the derivation to the same file.
+
+   :::tip
+   You may also create the derivation specification in a separate file; the results will be the same.
+   However, if using separate files, you must make sure that the file with the derivation imports the source collection's specification,
+   and that the top-level `flow.yaml` in your working directory imports all specification files.
+   [Learn more about imports.](../concepts/import.md)
+
+3. Add a new collection below the first one.
+
+   * The [schema](../concepts/schemas.md) and [collection key](../concepts/collections.md#keys) should reflect your desired output.
+   * Add the `derivation` stanza. The TypeScript module you name will be generated next, and you'll define the transformation's function there.
+
+   ```yaml
+   collections:
+      estuary/data-flow-name/greetings:
+        {...}
+      estuary/data-flow-name/dozen-greetings:
+         schema:
+            properties:
+               count:
+                  type: integer
+               message:
+                  type: string
+               dozens:
+                  type: integer
+            required:
+            - dozens
+            - count
+            - message
+            type: object
+         key:
+            - /dozens
+         derivation:
+            transform:
+               greetings-by-dozen:
+                  source: {name: estuary/data-flow-name/greetings}
+                  publish: {lambda: typescript}
+            typescript: {module: divide-by-twelve.ts}
+      ```
+   Save the file.
```

Comment:
I would maybe add some more description of what's going on here. Why did you add `dozens` to the `required` keys? Does that come from the `key: - /dozens` field? What does that field actually do? Do I _have_ to add it? Why? etc

Reply:
Maybe these are explained in more detail somewhere else in the docs? Maybe like a spec for this yaml document, explaining what various fields/stanzas do? If so we should link there

Reply:
I had linked schema and collection key docs above, but I can see how that'd be easy to just skim past. Added comments inline in the sample and I think it's a lot clearer now